crates/polars-io/src/ipc/ipc_reader_async.rs

```diff
@@ -0,0 +1,282 @@
+        let footer_metadata =
+            self.store
+                .get_range(
+                    &self.path,
+                    file_size.checked_sub(FOOTER_METADATA_SIZE).ok_or_else(|| {
+                        to_compute_err("ipc file size is smaller than the minimum")
+                    })?..file_size,
+                )
+                .await?;
+
+        let footer_size = deserialize_footer_metadata(
+            footer_metadata
+                .as_ref()
+                .try_into()
+                .map_err(to_compute_err)?,
+        )?;
+
+        let footer = self
+            .store
+            .get_range(
+                &self.path,
+                file_size
+                    .checked_sub(FOOTER_METADATA_SIZE + footer_size)
+                    .ok_or_else(|| {
+                        to_compute_err("invalid ipc footer metadata: footer size too large")
+                    })?..file_size,
+            )
+            .await?;
+
+        arrow::io::ipc::read::deserialize_footer(
+            footer.as_ref(),
+            footer_size.try_into().map_err(to_compute_err)?,
+        )
+    }
+
+    // pub async fn metadata(&self) -> PolarsResult<MappedMutexGuard<'_, FileMetadata>> {
+    //     let mut ipc_metadata = self.ipc_metadata.lock().await;
+    //     if ipc_metadata.is_none() {
+    //         *ipc_metadata = Some(self.fetch_ipc_metadata().await?)
+    //     }
+    //     Ok(MutexGuard::map(ipc_metadata, |x| {
+    //         x.as_mut().unwrap_or_else(|| unreachable!())
+    //     }))
+    // }
+
+    // TODO: Utilize previously obtained metadata.
+    pub async fn data(
+        &self,
+        metadata: Option<&FileMetadata>,
+        options: IpcReadOptions,
+        verbose: bool,
+    ) -> PolarsResult<DataFrame> {
+        let bytes = self.store.get(&self.path).await?;
+
+        let projection = match options.projection.as_deref() {
+            Some(projection) => {
+                fn prepare_schema(mut schema: Schema, row_index: Option<&RowIndex>) -> Schema {
+                    if let Some(rc) = row_index {
+                        let _ = schema.insert_at_index(0, rc.name.as_str().into(), IDX_DTYPE);
+                    }
+                    schema
+                }
+
+                // Retrieve the metadata for the schema so we can map column names to indices.
+                let fetched_metadata;
+                let metadata = if let Some(metadata) = metadata {
+                    metadata
+                } else {
+                    // This branch is  happens when _metadata is None, which can happen if we Deserialize the execution plan.
+                    fetched_metadata = self.metadata().await?;
+                    &fetched_metadata
+                };
+
+                let schema = prepare_schema((&metadata.schema).into(), options.row_index.as_ref());
+
+                let hive_partitions = None; // TODO
```

Comment:
Probably need to feed this in somehow. Not sure how it relates to single/multiple paths.

Reply:
@ritchie46 can you help me understand hive partitions and how to fill this value correctly if needed?

Reply:
Hive can be removed for now. We should first better design that.

Reply:
Does that mean I can leave this as is?